cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -241,8 +248,16 @@ function move_sriov_pfs_netns_to_node {
 
     ip link set "$pf_name" netns "$node"
     pf_array+=("$pf_name")
+    if [ "${#pf_array[@]}" -eq "$num_pfs_required" ]; then
+      break
+    fi
   done
 
+  if [ "${#pf_array[@]}" -lt "$num_pfs_required" ]; then
+    echo "FATAL: Could not allocate enough PFs, please check PF_BLACKLIST, NUM_PFS_REQUIRED, and how many PF actually available or used already"
+    exit 1
+  fi
+
```

Comment:
Instead of this check I would add check if there are enough PF's before the loop starts by comparing between `sriov_pfs length` and `num_pfs_required`, if false stop the script with informative message.
And same principal as before, echo "Moving PF's to netns" so in case of an error during moving the PF to a netns the script would stop and the error from iplink would reflect in the logs.

Reply:
It will need to take into account the black list ones,
I find it more complex, and imo not needed.
I think the current method is straightforward and simple.
The whole operation takes a jiffy.

Again i am against putting an anchor echo.